app/assets/javascripts/application.js.erb

```diff
@@ -11,3 +13,55 @@ window.addEventListener("DOMContentLoaded", () => {
     window['ga-disable-' + GA_ID] = true;
   }
 });
+
+const getPasswordStrength = password => {
+  const result = zxcvbn(password);
+  const score = 100 * result.score / 4;
+  const scoreText = score < 50 ? 'weak' : (score < 100 ? 'medium' : 'good');
+  const rawSuggestion = result.feedback.suggestions.join(', ');
+  // zxcvbn-js sometimes fails to return a suggestion even when the password is too weak, so we need to perform some coercion.
+  const suggestion = (!rawSuggestion) ? 'Choose a password that is harder to guess' : rawSuggestion;
+  return { score, scoreText, suggestion };
+};
+
+if ($('input[name="user[password]"]').length) {
+  $('body').on('keyup', 'input[name="user[password]"]', function(e) {
```

Comment:
The idea is to show the password strength *while* a user types their password, rather than typing and then clicking outside the password field just to update its strength, which is not a very intuitive experience. This pattern is particularly useful in our specific use-case, since we require quite strong passwords which would probably take a few tries to get right and we ought to make it as easy as possible.

This is also how it's been implemented on most other website (e.g. Google and Dropbox), but happy to continue the discuss if you'd like.